### ISSUE TITLE
Usability enhancements to Configuration GUI

### ISSUE THREAD

**Issue by aybe** [joncampbell123/dosbox-x#1847]
**Describe the solution you'd like**

Some usability issue with the config GUI:

### Use caps/good naming appropriately for clarity:

- SDL, MIDI, DOS, PC-98, IPX, etc...

![dosbox-x_2020-09-11_07-03-16](https://user-images.githubusercontent.com/1537591/92863973-e2c39c80-f3fc-11ea-9959-99ede293f84f.png)

### Sort options alphabetically:

Right now it's extremely difficult to spot the right option you need, they aren't sorted at all:

![dosbox-x_2020-09-11_07-05-50](https://user-images.githubusercontent.com/1537591/92864320-3df58f00-f3fd-11ea-86a6-d934745d1454.png)

**Comment by Wengier** [comment]
I agree! Section names like SDL, MIDI, DOS, PC-98, IPX can be capitalized, and some sorting is good too. By the way, in the latest code there is a "Show advanced options" checkbox which if unchecked will only show common config options (instead of all of them), which probably make it easier to use for normal users

**Comment by aybe** [comment]
Yesterday I tried to take a look at that and by doing so, I suddenly remembered how much of a mess is the code base.

- changing `section.GetName` won't work as it's used for both display and config
- `autoexec` is a special case
- litterals everywhere, just for the sake of easiness people have done insane copy/paste
  - some extra header would have been so much smarter, write once/reuse
  - replacing some things needs hundreds of replacements
- and it's not C# world, sometimes, tiny changes trigger lot of recompilation

Extremely frustrating then I abandoned because it already was very late.

**Comment by Wengier** [comment]
@aybe It is possible to separate the name for display and config, and I have already done this in the latest code. You can see the screenshots below:

![image](https://user-images.githubusercontent.com/8216923/92963860-2213dc80-f441-11ea-839c-3405c7ddffa6.png)

![image](https://user-images.githubusercontent.com/8216923/92963937-44a5f580-f441-11ea-855b-d3b4b09edf87.png)

**Comment by Wengier** [comment]
Perhaps section names like "Ide primary" can be capitalized to "IDE Primary" too.

**Comment by Wengier** [comment]
A new screenshot with section names like "Ide, primary" capitalized as well:

![image](https://user-images.githubusercontent.com/8216923/92965624-1d045c80-f444-11ea-8f22-179f6a2e1cfa.png)

**Comment by aybe** [comment]
Great !

What about
- `PC-98`
- `V-Sync`
- `SB` or `SB/SBPRO/SB16` ?
- `PC Speaker`
- maybe `COM`, `LPT` ?
- `MS-DOS`
- `IDE port 1` or `IDE #1` or `Hard Disk 1`, `Hard Drive 1`
- `FDC` or `Floppy Disk`
- `4DOS`
- `CONFIG.SYS`
- `AUTOEXEC.BAT`

**Comment by Wengier** [comment]
Good suggestions! How about this:

![image](https://user-images.githubusercontent.com/8216923/92973035-4c6d9600-f451-11ea-954a-3eb5addf636e.png)

**Comment by aybe** [comment]
I've been scratching my head trying to find a better layout : 

Now using columns instead ! I find it much easier to read.

![devenv_2020-09-12_01-57-41](https://user-images.githubusercontent.com/1537591/92981735-751a7d80-f49b-11ea-9a56-173c8413f31b.png)

As you can see I tried to group by topic/category...

Let me know what you think.

**Comment by Wengier** [comment]
@aybe Nice looking! It is certainly a good improvement over the original.
